### DISCUSSION TITLE
React Native - Return to dapp after approving session

### DISCUSSION THREAD

**Discussion by CallbackDan** [WalletConnect/walletconnect-monorepo#3134] [Developer Support] [answered]
Hi there,

Is there a way using the web3wallet package in React Native, to redirect back to the dapp in the browser, after approving a session proposal or session request?

So on mobile, the user is using visiting a dapp in their browser, taps connect with wc, deeplinks into app, approves the session and gets redirected back to the dapp. How can the redirect happen? 

For iOS it looks like there's a 'goBack' method, but doesn't seem to exist in the web3wallet package. More info here: https://docs.walletconnect.com/2.0/ios/router/installation

Thanks in advance

**Comment by ignaciosantise** [comment, marked as answer]
hey @CallbackDan 👋 we don't have a tool to do that in react native yet, but some wallets use a lib called `react-native-minimizer` to do that. It's not the best tool, as you'll probably have to patch things to make it work. You can check Rainbow repo, as they use it

**Reply by CallbackDan** [reply]
Thanks @ignaciosantise ! I'll check it out!

**Reply by Dmitriy555244** [reply]
![qr-merlin:0x47889D17373b3ed35eCcF51F38212682Bad9Ee97?amount=352](https://github.com/user-attachments/assets/1e2871e6-9a2d-43e1-a158-93c598b62408)

**Comment by adnansudozai** [comment]
@CallbackDan I also have face this issue can you solve this?
